### ISSUE TITLE
 Confusing error message for missing/not writable wp-rocket-config dir

### ISSUE THREAD

**Issue by jbma** [wp-media/wp-rocket#59]
This error message seems to appear when wp-rocket-config is not present inside wp-content, or when it is present, but not writeable:

https://bitbucket.org/repo/z9nBad/images/4059759762-Bildschirmfoto%202015-12-24%20um%2009.48.47.png

> Be careful, you don’t have writing permissions on WP Rocket domain configuration folder (wp-content/wp-rocket-config). To make WP Rocket work properly, please CHMOD 755 or 775 or 777 this folder. When the problem is solved, thank you to save the WP Rocket options to generate the configuration file.

I’m seeing several issues here:
1. Message is not clear about what is the problem.
The message says $user doesn’t have writing permissions on a particular directory, even though in reality that directory possibly doesn’t even exist yet. While it may sound like a legitimate shortcut to just talk about missing writing permissions, it apparently does confuse $user when they see that the directory in question doesn’t even exist.
I’d suggest WPR’s language should be more clear here:
Are writing permissions missing for an existing directory?
Or is that directory missing entirely?
…which would, in fact, lead us to 2 separate messages for 2 separate cases.
Accordingly, the next phrase “To make WP Rocket work properly” should follow the previously made distinction:
In order to enable WP Rocket to write to that folder…
In order to enable WP Rocket to create that folder…
1. Message may sound intimidating to non-tech-savvy $users.
   “please CHMOD 755 or 775 or 777 this folder” sounds a bit intimidating to me for two reasons:
   It is assumed $user knows what CHMOD means.
   It is assumed $user knows how to decide which of the 3 optional values they should apply.
   It’s obvious that we can not and should not try to educate people on Apache basics in our plugin UI. wink So why not go for a more concise directive here, and free $user from digging into tech entirely in case they’re not already firm with setting file permissions?
   Make our UI give 1 clear instruction (probably 777 in this case as it’s most likely to work). This is for tech-savvy $users.
   Remove the link on “writing permissions”. Instead append a different link at the end of the message for $users to open a support ticket for this very issue. Maybe when $user would click that link, they could be directed to the support tab and see a pre-defined message be filled in the form already?
   The message (for missing permissions) could look something like this then:

> Apparently WP Rocket does not have writing permissions for its domain configuration folder. Please CHMOD 777 the wp-content/wp-rocket-config folder, then go to Settings → WP Rocket and press Save once. (I’m not sure how to set writing permissions on a folder, please fix this issue for me.)

In closing, I’m aware that inviting $users to open a support ticket is not exactly what we’d like to do in order to reduce our amount of tickets. On the other hand, non-tech-savvy $users will end up opening a ticket here anyhow.
So I’d suggest we try keeping them happy directly in our UI, instead of confronting them with an intimidating UI first, and then try making them happy again afterwards at our help desk.

**Comment by Tabrisrp** [comment]
Wording for this has been updated in 2.11 to make it more user friendly and link to a complete documentation page.

**Comment by dayrisesolar** [comment]
I got this error message "WP Rocket cannot configure itself due to missing writing permissions. Affected file/folder: wp-content/advanced-cache.php"
I tried to change file permissions as suggested online forums but no result.  Finally I checked up the content of advanced-cache.php and found that WP Rocket is not defined properly. In my further search I found that probably WP Optimise plugin is conflicting.  As soon as I disabled WP Optimize (another caching plugin), the error got rectified. No need to change any file permission.  The error signal is wrong.
